x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/logging/Throttler.java

```diff
@@ -119,41 +116,38 @@ private static class LogExecutor {
         private final long skippedLogCalls;
         private final Instant timeOfLastLogCall;
         private final Clock clock;
-        private final Runnable logRunner;
-
-        LogExecutor(Clock clock, Consumer<String> logAppendedMessage) {
-            skippedLogCalls = 0;
-            timeOfLastLogCall = Instant.now(clock);
-            this.clock = clock;
-            // The first log message can log the original message without waiting
-            this.logRunner = () -> logAppendedMessage.accept("");
+        private final Consumer<String> consumer;
+
+        LogExecutor(Clock clock, Consumer<String> throttledConsumer) {
+            this(clock, 0, throttledConsumer);
         }
 
-        LogExecutor(Clock clock, long skippedLogCalls, Runnable logRunner) {
+        LogExecutor(Clock clock, long skippedLogCalls, Consumer<String> consumer) {
             this.skippedLogCalls = skippedLogCalls;
-            timeOfLastLogCall = Instant.now(clock);
-            this.clock = clock;
-            this.logRunner = logRunner;
+            this.clock = Objects.requireNonNull(clock);
+            timeOfLastLogCall = Instant.now(this.clock);
+            this.consumer = Objects.requireNonNull(consumer);
         }
 
-        void log() {
-            this.logRunner.run();
+        void log(String message) {
+            this.consumer.accept(message);
         }
 
         LogExecutor compute(Consumer<String> executor, Duration durationToWait) {
             if (hasDurationExpired(durationToWait)) {
-                String msg = "";
+                String messageToAppend = "";
                 if (this.skippedLogCalls == 1) {
-                    msg = ", repeated 1 time";
+                    messageToAppend = ", repeated 1 time";
                 } else if (this.skippedLogCalls > 1) {
-                    msg = format(", repeated %s times", this.skippedLogCalls);
+                    messageToAppend = format(", repeated %s times", this.skippedLogCalls);
                 }
 
-                String finalMsg = msg;
-                return new LogExecutor(this.clock, 0, () -> executor.accept(finalMsg));
+                final String stringToAppend = messageToAppend;
+                return new LogExecutor(this.clock, 0, (message) -> executor.accept(message.concat(stringToAppend)));
```

Comment:
This now handles doing the concatenation instead of having the original caller do it.